apps/site/assets/ts/schedule/components/ScheduleDirection.tsx

```diff
@@ -202,6 +216,44 @@ const ScheduleDirection = ({
     [route, state.directionId, busVariantId, currentRoutePatternIdForData]
   );
 
+  const channel = `vehicles:${route.id}:${state.directionId}`;
+
+  const [channelState, channelDispatch] = useReducer(channelReducer, {
+    channel,
+    markers: mapState.data ? mapState.data.markers : {}
+  });
+  useEffect(
+    () => {
+      setupChannels(channel, channelDispatch);
+      return () => stopChannels(channel);
+    },
+    [channel, channelDispatch]
+  );
+  const stopMarkers =
+    mapState.data && mapState.data.stop_markers
+      ? mapState.data.stop_markers
+          .filter((mark: MapMarker) => currentStops.includes(mark.id as string))
+          .map((marker: MapMarker) => updateMarker(marker))
+      : [];
+
+  const mapDataFromChannel = {
+    ...mapState.data,
+    polylines:
+      mapState.data && mapState.data.polylines
+        ? mapState.data.polylines.filter((p: Polyline) =>
+            currentShapes.some(shape => shape === p.id)
+          )
+        : [],
+    markers: channelState.markers.concat(stopMarkers)
+  };
+  const bounds = useRef(getBounds(stopMarkers));
+
+  const vehicleMarkers = mapDataFromChannel.markers
+    ? mapDataFromChannel.markers.filter(
+        (marker: MapMarker) => marker.icon === "vehicle-bordered-expanded"
```

Comment:
You had added an `isVehicleMarker` function elsewhere that you could totally use here 😉

Reply:
But why not use `channelState.markers` for the vehicle markers?